ooni/nettest.py

```diff
@@ -171,6 +171,9 @@ class NetTestLoader(object):
     def __init__(self, options, test_file=None, test_string=None,
                  annotations={}):
         self.options = options
+        # Ensure annotations are always dicts
+        if not isinstance(annotations, dict):
+            annotations = {}
```

Comment:
There are actually two catches here

- [ ] using mutable object as a default argument
```
class Foo(object):
    def __init__(self, annotations={}):
        self.annotations = annotations
        self.annotations['foo'] = self.annotations.get('foo', 0) + 1
        print self.annotations
a, b, c = Foo(), Foo(), Foo()
```

- [ ] dropping the data silently. IMHO, it should be something like `if ann is None: ann = {}; if not isinstance(ann, dict): throw TypeError`